### PR TITLE
HBASE-22904 NPE occurs when RS send space quota usage report during HMaster init

### PR DESCRIPTION
**Pull request by pankaj72981**
During HMaster failover, if RegionServer send space quota report to HMaster and MasterQuotaManager is not yet initialized then NPE will occur.

https://github.com/apache/hbase/blob/767bb1511285ae67dae15e8a2aab62c38244c797/hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java#L2229

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 107 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 418 | master passed |
| +1 | compile | 80 | master passed |
| +1 | checkstyle | 109 | master passed |
| +1 | shadedjars | 375 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 52 | master passed |
| 0 | spotbugs | 333 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 330 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 427 | the patch passed |
| +1 | compile | 64 | the patch passed |
| +1 | javac | 64 | the patch passed |
| +1 | checkstyle | 93 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 344 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1188 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 44 | the patch passed |
| +1 | findbugs | 298 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 13508 | hbase-server in the patch failed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 17977 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux ea33b424ba26 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / 767bb15112 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/1/testReport/ |
| Max. process+thread count | 4752 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by jatsakthi** [comment]
Thanks for the patch @pankaj72981 . Any UTs here?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 356 | master passed |
| +1 | compile | 55 | master passed |
| +1 | checkstyle | 78 | master passed |
| +1 | shadedjars | 284 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 40 | master passed |
| 0 | spotbugs | 259 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 255 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 305 | the patch passed |
| +1 | compile | 58 | the patch passed |
| +1 | javac | 58 | the patch passed |
| +1 | checkstyle | 77 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 285 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 966 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 38 | the patch passed |
| +1 | findbugs | 286 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 9282 | hbase-server in the patch passed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 12819 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 8b3be17f890d 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / 74fb2040ea |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/2/testReport/ |
| Max. process+thread count | 4804 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java [hunk 1]
```diff
@@ -2227,10 +2227,12 @@ public RegionSpaceUseReportResponse reportRegionSpaceUse(RpcController controlle
...
-      final long now = EnvironmentEdgeManager.currentTime();
-      for (RegionSpaceUse report : request.getSpaceUseList()) {
-        quotaManager.addRegionSize(ProtobufUtil.toRegionInfo(
-            report.getRegionInfo()), report.getRegionSize(), now);
+      if (quotaManager != null) {
+        final long now = EnvironmentEdgeManager.currentTime();
+        for (RegionSpaceUse report : request.getSpaceUseList()) {
+          quotaManager.addRegionSize(ProtobufUtil.toRegionInfo(report.getRegionInfo()),
+            report.getRegionSize(), now);
+        }
```
Comment (busbey) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java]: log a message that we got a report but were not yet ready and so discarded it. DEBUG level I think?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 211 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 332 | master passed |
| +1 | compile | 54 | master passed |
| +1 | checkstyle | 79 | master passed |
| +1 | shadedjars | 280 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 35 | master passed |
| 0 | spotbugs | 255 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 253 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 301 | the patch passed |
| +1 | compile | 56 | the patch passed |
| +1 | javac | 56 | the patch passed |
| +1 | checkstyle | 74 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 274 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 937 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 35 | the patch passed |
| +1 | findbugs | 254 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 18714 | hbase-server in the patch failed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 22282 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.master.procedure.TestSCPWithReplicas |
|   | hadoop.hbase.replication.TestReplicationKillSlaveRS |
|   | hadoop.hbase.namespace.TestNamespaceAuditor |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.replication.TestReplicationSmallTestsSync |
|   | hadoop.hbase.tool.TestSecureBulkLoadHFiles |
|   | hadoop.hbase.client.TestCloneSnapshotFromClientNormal |
|   | hadoop.hbase.replication.TestReplicationSmallTests |
|   | hadoop.hbase.tool.TestBulkLoadHFiles |
|   | hadoop.hbase.replication.TestReplicationSyncUpTool |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2faf256885d9 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / 2d4580106a |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/3/testReport/ |
| Max. process+thread count | 5210 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 89 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 513 | master passed |
| +1 | compile | 75 | master passed |
| +1 | checkstyle | 109 | master passed |
| +1 | shadedjars | 397 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 45 | master passed |
| 0 | spotbugs | 322 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 319 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 396 | the patch passed |
| +1 | compile | 67 | the patch passed |
| +1 | javac | 67 | the patch passed |
| +1 | checkstyle | 88 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 370 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1299 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 42 | the patch passed |
| +1 | findbugs | 329 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 19412 | hbase-server in the patch failed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 24088 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.tool.TestSecureBulkLoadHFiles |
|   | hadoop.hbase.master.procedure.TestProcedurePriority |
|   | hadoop.hbase.client.TestAsyncTableAdminApi |
|   | hadoop.hbase.namespace.TestNamespaceAuditor |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4d5665d78b01 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / 2d4580106a |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/4/testReport/ |
| Max. process+thread count | 4697 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 37 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 343 | master passed |
| +1 | compile | 58 | master passed |
| +1 | checkstyle | 78 | master passed |
| +1 | shadedjars | 291 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 35 | master passed |
| 0 | spotbugs | 246 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 244 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 315 | the patch passed |
| +1 | compile | 56 | the patch passed |
| +1 | javac | 56 | the patch passed |
| +1 | checkstyle | 77 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 288 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 971 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 35 | the patch passed |
| +1 | findbugs | 257 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8248 | hbase-server in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 11740 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 21387206c031 4.4.0-157-generic #185-Ubuntu SMP Tue Jul 23 09:17:01 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / 504fc52187 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/5/testReport/ |
| Max. process+thread count | 4758 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 319 | master passed |
| +1 | compile | 52 | master passed |
| +1 | checkstyle | 74 | master passed |
| +1 | shadedjars | 268 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 34 | master passed |
| 0 | spotbugs | 260 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 257 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 294 | the patch passed |
| +1 | compile | 51 | the patch passed |
| +1 | javac | 51 | the patch passed |
| +1 | checkstyle | 72 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 258 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 904 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 32 | the patch passed |
| +1 | findbugs | 264 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8895 | hbase-server in the patch passed. |
| +1 | asflicense | 23 | The patch does not generate ASF License warnings. |
| | | 12201 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux d84e8f8e8e00 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / ee9d98611b |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/9/testReport/ |
| Max. process+thread count | 4622 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/9/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 140 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 544 | master passed |
| +1 | compile | 121 | master passed |
| +1 | checkstyle | 161 | master passed |
| +1 | shadedjars | 501 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 93 | master passed |
| 0 | spotbugs | 350 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 348 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 330 | the patch passed |
| +1 | compile | 57 | the patch passed |
| +1 | javac | 57 | the patch passed |
| +1 | checkstyle | 80 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 287 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1006 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 36 | the patch passed |
| +1 | findbugs | 271 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 17031 | hbase-server in the patch failed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 21443 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestAsyncRegionAdminApi2 |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7e64413762da 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / ee9d98611b |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/7/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/7/testReport/ |
| Max. process+thread count | 4757 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/7/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 119 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 1 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 411 | master passed |
| +1 | compile | 55 | master passed |
| +1 | checkstyle | 90 | master passed |
| +1 | shadedjars | 292 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 36 | master passed |
| 0 | spotbugs | 263 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 261 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 313 | the patch passed |
| +1 | compile | 61 | the patch passed |
| +1 | javac | 61 | the patch passed |
| +1 | checkstyle | 90 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | shadedjars | 305 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 984 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 35 | the patch passed |
| +1 | findbugs | 243 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 16606 | hbase-server in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 20325 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.master.TestMasterShutdown |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/11/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 14b424541745 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / 94155fe175 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/11/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/11/testReport/ |
| Max. process+thread count | 4549 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/11/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by pankaj72981** [comment]
> Thanks for the patch @pankaj72981 . Any UTs here?

IMO UT is not required here, as we are throwing any exception and mocking is not useful.
Kindly let me know you opinion.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java (hunk 1 above)
Comment (pankaj72981) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java]: yeah, DEBUG message would be nice.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 415 | master passed |
| +1 | compile | 76 | master passed |
| +1 | checkstyle | 111 | master passed |
| +1 | shadedjars | 364 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 48 | master passed |
| 0 | spotbugs | 332 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 329 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 365 | the patch passed |
| +1 | compile | 76 | the patch passed |
| +1 | javac | 76 | the patch passed |
| +1 | checkstyle | 110 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 354 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 1216 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | javadoc | 44 | the patch passed |
| +1 | findbugs | 350 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 9484 | hbase-server in the patch passed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 13866 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/12/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/529 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 36f9841094c5 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-529/out/precommit/personality/provided.sh |
| git revision | master / 554b2531ca |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/12/testReport/ |
| Max. process+thread count | 4616 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-529/12/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**